src/lib/navigation/wizard-state.model.ts

```diff
@@ -92,6 +116,15 @@ export class WizardState {
     return this.wizardSteps.every(step => step.completed || step.optional);
   }
 
+  /**
+   * Step Changed Subject for notify and update when step changed
+   */
+  private stepChanged: Subject<WizardStep> = new Subject<WizardStep>();
```

Comment:
Why is an additional `Subject` required?
Isn't the output sufficient?

Reply:
I don’t think you can use `Output` in an Angular service

Reply:
You don’t you have limited options in purpose.

You only need 2 usages with this
1. Update step
2. Listen to updates

**Update step:**
When changing current index it will emit the value

**Listen to changes:**
You have a get variable/function for getting the observable

Reply:
Can't you then create the `Subject` in your service and wire the emitted step change to the subject by a provided function to the output?
I.e. like this:

```
stepChanged(event): void {
   this.service.subject.next(event);
}

<wizard (stepChanged)="stepChanged($event)">
   ...
</wizard>
```

Reply:
Where is the step changed function location? is it in the wizard.component.ts because you can create instance of yourself

Is the user have this function? Then we don’t provide him the subject only update and listening, can you please give me another example?

Reply:
And I think using an angular basic Tool is better instead of a workaround

Reply:
Here a more detailed example:

## The service
```typescript
@Injectable()
export class Service {
   public subject = new Subject();
}
```

## The application component
```
@Component({
   template: `
      <wizard (stepChanged)="stepChanged($event)">
         ...
      </wizard>
   `
})
export class AppComponent {
   constructor(private service: Service) {}

   public stepChanged(event): void {
      this.service.subject.next(event);
   }
}
```

Reply:
I agree, that adding functions directly to a library is often faster to implement and second easier to use from inside an application.

The issues for the library developers are, that each additional function needs to be documented and maintained.
In addition this easily leads to cluttering of the library, where a lot of functions are added to a library that are not often needed or which can be easily realized from outside of the library.

Reply:
So at least make the subject private and leave the functions for interact with the subject without changing the subject itself, this way you don’t open the subject for unwanted changes